lib/spack/spack/cmd/python.py

```diff
@@ -118,6 +118,9 @@ def python_interpreter(args):
     else:
         # Provides readline support, allowing user to use arrow keys
         console.push('import readline')
+        # Provide tabcompletion
+        console.push('from rlcompleter import Completer')
```

Comment:
Can you check if:
```suggestion
        console.push('import rlcompleter')
```
is sufficient to have completion (it seems so by reading Python docs)? Also, can you confirm this also works with Python 2.7?

Reply:
Thanks for having a look at this. I've checked that already, but this is only a half-measure and will only complete some things, like the contents of the spack module, but not newly defined objects.

I have also just checked, that this still works with Python 2.7

Reply:
Maybe to clarify further: What I described above as a half-measure is when *only* doing `import rlcompleter`, but not leaving out
```python
readline.set_completer(Completer(locals()).complete)
```
I could of course just import the whole module and do
```python
readline.set_completer(rlcompleter.Completer(locals()).complete)
```
if that is what you meant.

Reply:
Can you try:
```suggestion
        console.push('import rlcompleter')
        console.push('readline.parse_and_bind("tab: complete")')
```
? That should work too and that's what seems to be suggested in the Python docs [here](https://docs.python.org/3/library/rlcompleter.html)

Reply:
I have tried that too already :) It suffers from the same issue as just doing `import rlcompleter`, as quick example
```console
>>> spack.<tab> # completes fine
>>> a = [1,2,3]
>>> a.<tab> # does not complete
```
with the change in this PR both complete fine and this mimicks the behaviour of the regular Python interpreter (at least for the Python 3.9 on Debian 11 or spack installed Python 3.10.3).

Reply:
Ok, tried that and I see what you mean. I think though we miss:
```python
console.push('readline.parse_and_bind("tab: complete")')
```
Without that a <kbd>Tab</kbd> is just a regular <kbd>Tab</kbd> for me :slightly_smiling_face:

Reply:
Interesting! For me it works without that, I'll have a quick look, whether adding that changes anything for me (which would be weird).

Reply:
Pushed the additional line.